pkg/urbit/vere/io/ames.c

```diff
@@ -503,99 +514,149 @@ _ames_etch_head(u3_head* hed_u, c3_y buf_y[4])
              ^ ((hed_u->mug_l & 0xfffff) << 11)
              ^ ((hed_u->rel_o &     0x1) << 31);
 
-  //  only version 0 currently recognized
-  //
-  c3_assert( 0 == hed_u->ver_y );  //  XX remove after testing
-
-  buf_y[0] = hed_w & 0xff;
-  buf_y[1] = (hed_w >>  8) & 0xff;
-  buf_y[2] = (hed_w >> 16) & 0xff;
-  buf_y[3] = (hed_w >> 24) & 0xff;
+  _ames_etch_word(buf_y, hed_w);
 }
 
-/* _ames_etch_prelude(): serialize packet prelude
+/* _ames_etch_prel(): serialize packet prelude
 */
-static c3_y
-_ames_etch_prelude(u3_head* hed_u, u3_prel* pre_u, c3_y* buf_y)
+static void
+_ames_etch_prel(u3_head* hed_u, u3_prel* pre_u, c3_y* buf_y)
 {
-  c3_y  rog_y = ( c3y == hed_u->rel_o ) ? 6 : 0;           //  origin len
-  c3_y  sen_y = 2 << pre_u->sic_y;
-  c3_y  rec_y = 2 << pre_u->ric_y;
-  c3_y  len_y = sen_y + rec_y + rog_y;
+  c3_w cur_w = 0;
 
-
-  // copy lives
   buf_y[0] = (pre_u->sic_y & 0xf) ^ ((pre_u->ric_y & 0xf) << 4);
+  cur_w++;
+
+  c3_y sen_y = 2 << hed_u->sac_y;
+  _ames_ship_of_chubs(pre_u->sen_d, sen_y, buf_y + cur_w);
+  cur_w += sen_y;
 
-  _ames_ship_of_chubs(pre_u->sen_d, sen_y, buf_y + 1);
-  _ames_ship_of_chubs(pre_u->rec_d, rec_y, buf_y + 1 + sen_y);
+  c3_y rec_y = 2 << hed_u->rac_y;
+  _ames_ship_of_chubs(pre_u->rec_d, rec_y, buf_y + cur_w);
+  cur_w += rec_y;
 
+  c3_y  rog_y = ( c3y == hed_u->rel_o ) ? 6 : 0;           //  origin len
   if ( rog_y ) {
```

Comment:
```suggestion
  c3_y rog_y = ( c3y == hed_u->rel_o ) ? 6 : 0;           //  origin len
  if ( rog_y ) {
```

Though I'm a little unclear on why this isn't just `if ( c3y == hed_u->rel_o )`.

Also I think margin comments (outside of types/structs) are out of style in vere?

Reply:
cleaned up